lib/optimizely/event_dispatcher.rb

```diff
@@ -28,26 +30,46 @@ class EventDispatcher
     # @api constants
     REQUEST_TIMEOUT = 10
 
+    def initialize(logger: nil, error_handler: nil)
+      @logger = logger || NoOpLogger.new
+      @error_handler = error_handler || NoOpErrorHandler.new
```

Comment:
I'm confused why the keyword params are defaulted to nil, only to have that default overridden?

I think the idiomatic way to do this would be:
```
def initialize(logger: NoOpLogger.new, error_handler: NoOpErrorHandler.new)
  @logger = logger
  @error_handler = error_handler
end
```